js/ai/src/generate/action.ts

```diff
@@ -287,6 +317,16 @@ async function generate(
     tools: newTools,
     toolCoice: newToolChoice,
   };
+  if (interruptedParts.length > 0) {
+    return {
+      ...response.toJSON(),
+      finishReason: 'interrupted',
+      message: {
+        role: 'model',
+        content: interruptedParts,
+      },
+    };
+  }
```

Comment:
Will this work properly if the model had already generated some text before it made the tool request? It looks like it won't, but maybe I'm not understanding the codepath. The existing parts should be examined and the request swapped instead of only returning the interrupt I think.

Reply:
done